smartsim/_core/mli/workermanager.py

```diff
@@ -0,0 +1,976 @@
+
+
+class InferenceRequest:
+    """Temporary model of an inference request"""
+
+    def __init__(
+        self,
+        model_key: t.Optional[str] = None,
+        callback: t.Optional[CommChannel] = None,
+        raw_inputs: t.Optional[t.List[bytes]] = None,
+        input_keys: t.Optional[t.List[str]] = None,
+        output_keys: t.Optional[t.List[str]] = None,
+        raw_model: t.Optional[bytes] = None,
+        batch_size: int = 0,
+        device: t.Optional[str] = None,
+    ):
+        """Initialize the InferenceRequest"""
+        self.model_key = model_key
+        self.raw_model = raw_model
+        self.callback = callback
+        self.raw_inputs = raw_inputs
+        self.input_keys = input_keys or []
+        self.output_keys = output_keys or []
+        self.batch_size = batch_size
+        self.device = device
+
+    @staticmethod
+    def from_dict(source: t.Dict[str, t.Any]) -> "InferenceRequest":
+        return InferenceRequest(
+            # **source
+        )
+
+
+class InferenceReply:
+    """Temporary model of a reply to an inference request"""
+
+    def __init__(
+        self,
+        outputs: t.Optional[t.Collection[t.Any]] = None,
+        output_keys: t.Optional[t.Collection[str]] = None,
+        failed: bool = False,
+    ) -> None:
+        """Initialize the InferenceReply"""
+        self.outputs: t.Collection[t.Any] = outputs or []
+        self.output_keys: t.Collection[t.Optional[str]] = output_keys or []
+        self.failed = failed
+
+
+class ModelLoadResult:
+    """A wrapper around a loaded model"""
+
+    def __init__(self, model: t.Any) -> None:
+        """Initialize the ModelLoadResult"""
+        self.model = model
+
+
+class InputTransformResult:
+    """A wrapper around a transformed input"""
+
+    def __init__(self, result: t.Any) -> None:
+        """Initialize the InputTransformResult"""
+        self.transformed = result
+
+
+class ExecuteResult:
+    """A wrapper around inference results"""
+
+    def __init__(self, result: t.Any) -> None:
+        """Initialize the ExecuteResult"""
+        self.predictions = result
+
+
+class InputFetchResult:
+    """A wrapper around fetched inputs"""
+
+    def __init__(self, result: t.Any) -> None:
+        """Initialize the InputFetchResult"""
+        self.inputs = result
+
+
+class OutputTransformResult:
+    """A wrapper around inference results transformed for transmission"""
+
+    def __init__(self, result: t.Any) -> None:
+        """Initialize the OutputTransformResult"""
+        self.outputs = result
+
+
+class BatchResult:
+    """A wrapper around batched inputs"""
+
+    def __init__(self, result: t.Any) -> None:
+        """Initialize the BatchResult"""
+        self.batch = result
+
+
+class FetchModelResult:
+    """A wrapper around raw fetched models"""
+
+    def __init__(self, result: bytes) -> None:
+        """Initialize the BatchResult"""
+        self.model_bytes = result
+
+
+class MachineLearningWorkerCore:
```

Comment:
Should we move the ``MachineLearningWorkerCore`` and ``MachineLearningWorkerBase`` to a file like ``smartsim/_core/mli/ml_worker.py`` or something similiar?

Reply:
moved those to `mli/worker.py` but let's revisit and decide if that's good enough organization

Reply:
moved `MachineLearningWorkerCore` and `MachineLearningWorkerBase` to `mli/infrastructure/worker/worker.py`

Reply:
moved all mock/test implementations into `tests/mli/*.py`